psi4/src/psi4/libmints/orbitalspace.cc

```diff
@@ -206,79 +206,54 @@ OrbitalSpace orthogonal_compliment(const OrbitalSpace &space1, const OrbitalSpac
         return OrbitalSpace(id, name, space2.C(), space2.evals(), space2.basisset(), space2.integral());
     }
 
-    // O12 = O12
+    // Overlap Matrix
     SharedMatrix O12 = OrbitalSpace::overlap(space1, space2);
-    auto C12 = std::make_shared<Matrix>("C12", space1.C()->colspi(), space2.C()->colspi());
 
-    // C12 = C1t * S12 * C2
-    // TODO: Try using the svd_a function of Matrix, however it calls dgesdd not dgesvd.
-    C12->transform(space1.C(), O12, space2.C());
-    C12->print();
-
-#if SVD
-    std::tuple<SharedMatrix, SharedMatrix, SharedMatrix> svd_temps = C12->svd_a_temps();
-
-#else
-    // We're interested in the right side vectors (V) of an SVD solution.
-    // We don't need a full SVD solution just part of it. Do it by hand:
-    auto D11 = std::make_shared<Matrix>("D11", C12->colspi(), C12->colspi());
-    D11->gemm(true, false, 1.0, C12, C12, 0.0);
-    //        D11->print();
-
-    auto V11 = std::make_shared<Matrix>("V11", D11->rowspi(), D11->colspi());
-    auto E1 = std::make_shared<Vector>("E1", D11->colspi());
-    D11->diagonalize(V11, E1);
-    //        V11->eivprint(E1);
-
-    // Count the number of eigenvalues < lindep_tol
-    Dimension zeros(space1.nirrep());
-    for (int h = 0; h < space1.nirrep(); ++h) {
-        for (int i = 0; i < E1->dimpi()[h]; ++i) {
-            if (E1->get(h, i) < lindep_tol) zeros[h]++;
-        }
+    // Half-transform to RIBS
+    auto C12 = std::make_shared<Matrix>("C12", space1.C()->colspi(), space2.C()->colspi());
+    C12->gemm(false, false, 1.0, O12, space2.C(), 0.0);
+
+    // SVD of MO overlap matrix
+    std::tuple<SharedMatrix, SharedVector, SharedMatrix> svd_a_tmps = C12->svd_a_temps();
+    SharedMatrix U = std::get<0>(svd_a_tmps);
+    SharedVector S = std::get<1>(svd_a_tmps);
+    SharedMatrix Vt = std::get<2>(svd_a_tmps);
```

Comment:
Just at a glance, is there a reason not to use `std::tie` here?

I can do a more thorough review later.

Reply:
You're right, I've updated to an even shorter form within C++17 standard.